plugins/src/main/kotlin/io/github/yonggoose/organizationdefaults/ArtifactCheckPlugin.kt

```diff
@@ -2,93 +2,326 @@ package io.github.yonggoose.organizationdefaults
+            } else {
+                project.logger.warn("POM file not found at expected locations")
+            }
+        } else {
+            validatePomSignatureFile(project, pomFile, pomSignatureFile, errors)
+        }
+    }
+
+    private fun validatePomSignatureFile(
+        project: Project,
+        pomFile: File,
+        pomSignatureFile: File,
+        errors: MutableList<String>
+    ) {
+        val verified = verifyPgpSignatureBasic(pomFile, pomSignatureFile, project)
+        if (!verified) {
+            errors.add("PGP signature verification FAILED for POM file")
+        } else {
+            project.logger.info("PGP signature verified for POM file")
+        }
+    }
+
+    private fun findSignatureFileForArtifact(
+        project: Project,
+        artifactFile: File,
+        signatureArtifacts: PublishArtifactSet
+    ): File? {
+        val expectedSignatureName = "${artifactFile.name}.asc"
+
+        project.logger.info("Looking for ${expectedSignatureName}")
+        return signatureArtifacts.find { signatureArtifact ->
+            val signatureFile = signatureArtifact.file
+            signatureFile.name == expectedSignatureName ||
+                    signatureFile.name.startsWith(artifactFile.nameWithoutExtension) && signatureFile.name.endsWith(".asc")
+        }?.file
+    }
+
+    private fun verifyPgpSignatureBasic(artifactFile: File, signatureFile: File, project: Project): Boolean {
+        return try {
+            if (!artifactFile.exists()) {
+                project.logger.error("Artifact file does not exist: ${artifactFile.absolutePath}")
+                return false
+            }
+
+            if (!signatureFile.exists()) {
+                project.logger.error("Signature file does not exist: ${signatureFile.absolutePath}")
+                return false
+            }
+
+            if (signatureFile.length() == 0L) {
+                project.logger.error("Signature file is empty: ${signatureFile.absolutePath}")
+                return false
+            }
+
+            val signatureContent = signatureFile.readText()
+            if (!signatureContent.contains("BEGIN PGP SIGNATURE") ||
+                !signatureContent.contains("END PGP SIGNATURE")) {
+                project.logger.error("Invalid PGP signature format in: ${signatureFile.absolutePath}")
+                return false
+            }
+
+            project.logger.info("Basic PGP signature validation passed for: ${artifactFile.name}")
+
+            return verifyPgpSignatureWithBouncyCastle(artifactFile, signatureFile, project)
+        } catch (e: Exception) {
+            project.logger.error("Error during basic PGP signature verification for ${artifactFile.name}: ${e.message}")
+            false
+        }
+    }
+
+    private fun verifyPgpSignatureWithBouncyCastle(
+        artifactFile: File,
+        signatureFile: File,
+        project: Project
+    ): Boolean {
```

Comment:
I don't think this actually verifies the signature, right? Just the presence of it?

Suggestion to rename with `verifyPgpSignaturePresenceWithBouncyCastle`